### PR TITLE
filtering share content

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
this resolve issue reported in https://trac.openmicroscopy.org.uk/ome/ticket/12804

to test log in to trout as root and open share 1415. it should contain 1 image (you won't be able to see thumbnail), other objects are filtered now

check also integration test if passes

cc: @jburel @will-moore

### CODE REVIEW

**Comment by jburel** [comment]
restarted travis

**Comment by jburel** [comment]
Probably good to add some robot tests for share since more bugs have to  be fixed during 5.1.1

**Comment by aleksandra-tarkowska** [comment]
@jburel we will add robot tests once we have all bugs fixed to avoid rewriting them. I hope that make sense

**Comment by jburel** [comment]
I think it will be better to do it as we go since one bug fix could break another previous fix.

**Comment by aleksandra-tarkowska** [comment]
@jburel I am afraid we do not have appropriate setup for testing this particular case at this moment. I prefer to do it in the end or the share review consulting all changes we made. We know exactly what kind of setup we need. As far as I can tell we do not test shares at all.

**Comment by jburel** [comment]
* log as root
* Click on 1415 in tree under Public data
```
Traceback (most recent call last):

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/django/core/handlers/base.py", line 114, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/decorators.py", line 469, in wrapped
    retval = f(request, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/decorators.py", line 519, in wrapper
    context = f(request, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/views.py", line 2682, in load_public
    controller.loadShareContent()

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/controller/share.py", line 201, in loadShareContent
    for ex in content:

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/webclient_gateway.py", line 1656, in getContents
    yield obj

UnboundLocalError: local variable 'obj' referenced before assignment


<WSGIRequest
path:/merge/webclient/load_public/1415/,
GET:<QueryDict: {u'view': [u'icon']}>,
POST:<QueryDict: {}>,
COOKIES:{'_ga': 'GA1.2.939380321.1423738908',
 'csrftoken': 'DPNlNKCb3H1wZsfBuYOJSC3Q7AfEx9le',
 'phpbb3_a9mfv_k': '',
 'phpbb3_a9mfv_sid': '22362440bb31efe41ce650615660d3c6',
 'phpbb3_a9mfv_u': '1',
 'sessionid_merge': 'w4f36zqo9jbu1esj1a56o5tcz995omvf',
 'style_cookie': 'null'},
META:{'CONTENT_LENGTH': '',
 'CONTENT_TYPE': '',
 u'CSRF_COOKIE': u'DPNlNKCb3H1wZsfBuYOJSC3Q7AfEx9le',
 'HTTPS': 'on',
 'HTTP_ACCEPT': 'text/html, */*; q=0.01',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate, sdch',
 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.8,fr;q=0.6',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_COOKIE': '_ga=GA1.2.939380321.1423738908; phpbb3_a9mfv_u=1; phpbb3_a9mfv_k=; phpbb3_a9mfv_sid=22362440bb31efe41ce650615660d3c6; style_cookie=null; csrftoken=DPNlNKCb3H1wZsfBuYOJSC3Q7AfEx9le; sessionid_merge=w4f36zqo9jbu1esj1a56o5tcz995omvf',
 'HTTP_HOST': 'trout.openmicroscopy.org',
 'HTTP_REFERER': 'https://trout.openmicroscopy.org/merge/webclient/public/',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36',
 'HTTP_X_REQUESTED_WITH': 'XMLHttpRequest',
 'PATH_INFO': u'/webclient/load_public/1415/',
 'QUERY_STRING': 'view=icon',
 'REQUEST_METHOD': 'GET',
 'SCRIPT_INFO': '/merge',
 'SCRIPT_NAME': u'/merge',
 'SERVER_NAME': 'ome-c6100-3',
 'SERVER_PORT': '443',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'wsgi.errors': <flup.server.fcgi_base.TeeOutputStream object at 0x52a2050>,
 'wsgi.input': <flup.server.fcgi_base.InputStream object at 0x52a2290>,
 'wsgi.multiprocess': True,
 'wsgi.multithread': False,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'https',
 'wsgi.version': (1, 0)}>
```

**Comment by jburel** [comment]
@aleksandra-tarkowska  just noticed that Travis failed so it was not included

**Comment by jburel** [comment]
Crash reported no longer occurs

**Comment by jburel** [comment]
copyright header in test_share.py needs to be modified.

**Comment by aleksandra-tarkowska** [comment]
@jburel done

File: components/tests/python/library/__init__.py [hunk 1]
```diff
@@ -214,6 +214,12 @@ def new_image(self, name=""):
         img.acquisitionDate = rtime(0)
         return img
 
+    def new_dataset(self, name="", description=None):
```
Comment (sbesson) [inline components/tests/python/library/__init__.py]: What's the reasoning for adding this helper method as opposed to using `ITest.make_dataset()` ?

Comment (aleksandra-tarkowska) [inline components/tests/python/library/__init__.py]: For better performance? With creating 2 objects, linking them and saving only link you make 1 db call rather then 3, right? In the future we could have one helper new_iobject(class, name) that allows you to create all type of subclass of ObjectI and one make_object(class, name) to create and save.

**Comment by jburel** [comment]
Some refactoring required while working on tests during the course of 5.1.x
This can be done in follow-up PRs.

**Comment by aleksandra-tarkowska** [comment]
Thank you everyone for helping

**Pull request merged**